src/coaster.h

```diff
@@ -53,13 +53,28 @@ class CarType {
 		return this->cars[(uint)(pitch & 0xf) | ((uint)(roll & 0xf) << 4) | ((uint)(yaw & 0xf) << 8)];
 	}
 
-	ImageData *cars[4096];   ///< Images of the car, in all possible orientation (if available).
+	/**
+	 * Retrieve a guest overlay sprite at a given \a pitch, \a roll, and \a yaw orientation.
+	 * @param pitch Required pitch of the car.
+	 * @param roll Required roll of the car.
+	 * @param yaw Required yaw of the car.
+	 * @param slot Seat number of the guest to overlay.
+	 * @return The overlay in the requested orientation, if available.
+	 */
+	const ImageData *GetGuestOverlay(uint pitch, uint roll, uint yaw, uint slot) const
+	{
+		return this->guest_overlays[slot * 16u*16u*16u + ((uint)(pitch & 0xf) | ((uint)(roll & 0xf) << 4) | ((uint)(yaw & 0xf) << 8))];
```

Comment:
The more modern style of these casts is to use `std::static_cast<T>(value)`, although I would guess there are loads of old-style casts in the code too.